metadata/src/main/java/org/apache/kafka/metadata/migration/KRaftMigrationDriver.java

```diff
@@ -88,14 +83,23 @@ private void initializeMigrationState() {
     }
 
     private boolean isControllerQuorumReadyForMigration() {
-        // TODO
-        return false;
+        return true;
     }
 
     private boolean areZkBrokersReadyForMigration() {
-        // TODO: Check available broker registrations and known topic assignments to confirm of
-        //  all Zk brokers are registered before beginning migration.
-        return false;
+        Set<Integer> kraftRegisteredZkBrokers = image.cluster().brokers().values()
+            .stream()
+            .filter(BrokerRegistration::isMigratingZkBroker)
+            .map(BrokerRegistration::id)
+            .collect(Collectors.toSet());
+        Set<Integer> zkRegisteredZkBrokers = zkMigrationClient.readBrokerIds();
+        zkRegisteredZkBrokers.removeAll(kraftRegisteredZkBrokers);
+        if (zkRegisteredZkBrokers.isEmpty()) {
+            return true;
+        } else {
```

Comment:
I forgot about the approach we decided to take. Remind me again, are we not looking at topic assignments at this point? I thought that is more decisive regarding the zkBrokers which need to register with the kraft controller.

Reply:
I'm tenatively adding the change. We can remove it if it is not required.